src/iterators/union_iterator.c

```diff
@@ -0,0 +1,408 @@
+/*
+ * Copyright Redis Ltd. 2016 - present
+ * Licensed under your choice of the Redis Source Available License 2.0 (RSALv2) or
+ * the Server Side Public License v1 (SSPLv1).
+ */
+
+#include "union_iterator.h"
+
+static inline int cmpLastDocId(const void *e1, const void *e2, const void *udata) {
+  const QueryIterator *it1 = e1, *it2 = e2;
+  return (int64_t)(it2->lastDocId - it1->lastDocId);
```

Comment:
safe?

Reply:
only if we can cast the diff to a positive `int64_t` (diff is less than 2^63). I'll change to a safer check even though this should be enough